plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -103,4 +103,83 @@ public void testPropertiesTable()
             assertUpdate("DROP TABLE IF EXISTS " + tableName);
         }
     }
+
+    @Test
+    public void testPartitionsTable()
+    {
+        String tableName = "test_simple_partitions_table";
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (_bigint BIGINT, _date DATE) WITH (partitioned_by = ARRAY['_date'] )");
+            assertUpdate("INSERT INTO " + tableName + " VALUES (0, CAST('2019-09-08' AS DATE)), (1, CAST('2019-09-09' AS DATE)), (2, CAST('2019-09-09' AS DATE))", 3);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (3, CAST('2019-09-09' AS DATE)), (4, CAST('2019-09-10' AS DATE)), (5, CAST('2019-09-10' AS DATE))", 3);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (6, NULL)", 1);
+            assertQuery("SELECT count(*) FROM " + tableName, "VALUES 7");
+            assertQuery("SELECT count(*) FROM \"" + tableName + "$partitions\"", "VALUES 4");
+
+            assertQuery("SHOW COLUMNS FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('partition', 'row(_date date)', '', ''),
+                            ('file_count', 'bigint', '', ''),
+                            ('total_size', 'bigint', '', '')
+                            """);
+
+            assertQuery(
+                "SElECT partition._date FROM \"" + tableName + "$partitions\"", " VALUES (DATE '2019-09-08'), (DATE '2019-09-09'), (DATE '2019-09-10'), (NULL)");
+        }
+        finally{
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
+
+    @Test
+    public void testPartitionsTableMultipleColumns()
+    {
+        String tableName = "test_partitions_table_multiple_columns";
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (_bigint BIGINT, _date DATE, _varchar VARCHAR) WITH (partitioned_by = ARRAY['_date', '_varchar'] )");
+            assertUpdate("INSERT INTO " + tableName + " VALUES (0, CAST('2019-09-08' AS DATE), 'a'), (1, CAST('2019-09-09' AS DATE), 'b'), (2, CAST('2019-09-09' AS DATE), 'c')", 3);
```

Comment:
let's add as well `null` values for the partition values.

Reply:
Resolved